torch/nn/parallel/distributed.py

```diff
@@ -314,6 +314,20 @@ def _ddp_init_helper(self):
         (4) registering the grad hooks
         (5) passing a handle of DDP to SyncBatchNorm Layer
         """
+
+        def parameters(m, recurse=True):
+            def model_parameters(m):
+                ps = m._former_parameters.values() \
+                    if hasattr(m, "_former_parameters") \
+                    else m.parameters(recurse=False)
+                for p in ps:
+                    yield p
+
+            ms = m.modules() if recurse else [m]
```

Comment:
what is this for? looks like it is not used?

Reply:
my bad will update.